cmd/fs-v1.go

```diff
@@ -353,17 +550,41 @@ func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.
 	if err = checkPutObjectArgs(bucket, object, fs); err != nil {
 		return ObjectInfo{}, err
 	}
+
 	// No metadata is set, allocate a new one.
 	if metadata == nil {
 		metadata = make(map[string]string)
 	}
 
-	uniqueID := mustGetUUID()
+	var fsMeta fsMetaV1
+	var wlk *lock.RWLockedFile
+	if bucket != minioMetaBucket {
+		// Save objects' metadata in `fs.json`.
+		// Skip creating fs.json if bucket is .minio.sys as the object would have been created
+		// by minio's S3 layer (ex. policy.json)
+		fsMeta = newFSMetaV1()
+		fsMeta.Meta = metadata
+
+		fsMetaPath := pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix, bucket, object, fsMetaJSONFile)
+		wlk, err = fs.rwPool.Create(fsMetaPath)
```

Comment:
The reason it is kept is for consistency since it was easy to read Open,Create -  right now there is no such need.